src/network.tf

```
@@ -27,3 +27,204 @@ module "vnet_integration" {
   tags = var.tags
 }
 
+
+## Application gateway public ip ##
+resource "azurerm_public_ip" "apigateway_public_ip" {
+  name                = format("%s-appgateway-pip", local.project)
+  resource_group_name = azurerm_resource_group.rg_vnet.name
+  location            = azurerm_resource_group.rg_vnet.location
+  sku                 = "Standard"
+  allocation_method   = "Static"
+
+  tags = var.tags
+}
+
+# Subnet to host the application gateway
+module "appgateway-snet" {
+  source               = "git::https://github.com/pagopa/azurerm.git//subnet?ref=v1.0.7"
+  name                 = format("%s-appgateway-snet", local.project)
+  address_prefixes     = var.cidr_subnet_appgateway
+  resource_group_name  = azurerm_resource_group.rg_vnet.name
+  virtual_network_name = module.vnet.name
+}
+
+## Application gateway ## 
+# Since these variables are re-used - a locals block makes this more maintainable
+locals {
+  backend_address_pool_name       = format("%s-appgw-be-address-pool", local.project)
+  frontend_http_port_name         = format("%s-appgw-fe-http-port", local.project)
+  frontend_https_port_name        = format("%s-appgw-fe-https-port", local.project)
+  frontend_ip_configuration_name  = format("%s-appgw-fe-ip-configuration", local.project)
+  http_setting_name               = format("%s-appgw-be-http-settings", local.project)
+  http_listener_name              = format("%s-appgw-fe-http-settings", local.project)
+  https_listener_name             = format("%s-appgw-fe-https-settings", local.project)
+  http_request_routing_rule_name  = format("%s-appgw-http-reqs-routing-rule", local.project)
+  https_request_routing_rule_name = format("%s-appgw-https-reqs-routing-rule", local.project)
+  acme_le_ssl_cert_name           = format("%s-appgw-acme-le-ssl-cert", local.project)
+  http_to_https_redirect_rule     = format("%s-appgw-http-to-https-redirect-rule", local.project)
+}
+
+# Application gateway: Multilistener configuraiton
+module "app_gw" {
+  source = "./modules/app_gw"
```

```suggestion
  source = "git::https://github.com/pagopa/azurerm.git//app_gateway?ref=v1.0.52"
```
point to new common azurerm module and delete ./modules dir